torch/_inductor/codegen/wrapper.py

```diff
@@ -496,10 +496,28 @@ def generate_extern_kernel_out(self, output_view, codegen_reference, args, kerne
             args.append(f"out={codegen_reference}")
         self.writeline(f"{kernel}({', '.join(args)})")
 
-    def generate_user_defined_triton_kernel(self, kernel_name, grid, args):
+    def generate_user_defined_triton_kernel(self, kernel_name, grid, configs, args):
+        assert len(grid) != 0
+        if len(grid) == 1:
+            grid = f"grid({grid[0]})"
+        else:
+            grid_wrapper = IndentedBuffer()
+            grid_wrapper.writeline(f"def grid_wrapper_for_{kernel_name}(meta):")
+            assert len(grid) == len(configs)
+            with grid_wrapper.indent():
+                for i, (grid_val, conf) in enumerate(zip(grid, configs)):
+                    guards = [
+                        f"meta['{name}'] == {val}" for name, val in conf.kwargs.items()
+                    ]
+                    guards = " and ".join(guards)
+                    grid_wrapper.writeline(f"if {guards}: return grid({grid[i]})(meta)")
```

Comment:
Could we share code with the implementation above?  I think we can preprocess this into a dict, so the body is more like:
```
def grid(meta):
  return precomputed_grids[(meta["XBLOCK], meta["YBLOCK])]
```

Reply:
One edge case I see here is
```
@triton.autotune(
        configs=[
            triton.Config(
                {"BLOCK_SIZE_X": 128, "BLOCK_SIZE_Y": 128}, num_stages=3, num_warps=8
            ),
            triton.Config(
                {"BLOCK_SIZE_X": 64}, num_stages=3, num_warps=8
            ),
        ],
        key=[],
    )
```
where configs are not of equal size. Is this something I should disallow?

Reply:
I think we can just disallow this.